cms/datasets/models.py

```diff
@@ -34,24 +86,29 @@ class ONSDataset(APIModel):
     search_fields: ClassVar[list[str]] = ["title", "version", "formatted_edition"]
 
     class Meta:
-        base_url: str = settings.DATASETS_API_BASE_URL
-        detail_url: str = f"{settings.DATASETS_API_BASE_URL}/%s"
+        base_url: str = settings.DATASET_EDITIONS_API_URL
+        detail_url: str = f"{settings.DATASET_EDITIONS_API_URL}/%s"
         fields: ClassVar = ["id", "description", "title", "version", "edition"]
         pagination_style = "offset-limit"
         verbose_name_plural = "ONS Datasets"
 
     @classmethod
     def from_query_data(cls, data: Mapping) -> "ONSDataset":
-        url: str = data["links"]["latest_version"]["href"]
-        edition_match = EDITIONS_PATTERN.search(url)
-        if not edition_match:
-            raise ValueError(f"Found invalid dataset URL, missing edition: {url} for dataset: {data['id']}")
-        edition = edition_match.group(1)
+        # Handle new /v1/dataset-editions response structure
+        dataset_id = data.get("dataset_id", data.get("id", ""))
+        title = data.get("title", "Title not provided")
+        description = data.get("description", "Description not provided")
+        edition = data.get("edition", "")
+
+        # Extract version from latest_version object
+        latest_version = data.get("latest_version", {})
+        version_id = latest_version.get("id", "1") if isinstance(latest_version, dict) else "1"
```

Comment:
Like ID, if the edition or version doesn't exist in the response for some reason, this would be considered malformed. I suggest we skip it. We don't want the whole thing to collapse if there is a single faulty dataset. 

If we can't skip it easily due to queryish's limitations, I'm happy for us to consider defaulting bad values as an MVP.